### PR TITLE
[Version] Increment to 3.0

### PR DESCRIPTION
**Pull request by kavilla**
### Description
Increment from 2.0.0 to 3.0.0, also some version checks.
Also, add a release build verification workflow. To verify x64
and ARM64 artifacts are built successfully with dependencies.

Signed-off-by: Kawika Avilla <kavilla414@gmail.com>
 
### Issues Resolved
n/a
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [x] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -106,3 +106,58 @@ jobs:
...
+      - name: Setup Node
+        uses: actions/setup-node@v2
+        with:
+          node-version-file: "./artifacts/.nvmrc"
+          registry-url: 'https://registry.npmjs.org'
+
+      - name: Setup Yarn
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
```
Comment (ashwin-pc) [inline .github/workflows/build_and_test_workflow.yml:142]: Do we need to specify version number?

File: .github/workflows/build_and_test_workflow.yml [hunk 2]
```diff
@@ -106,3 +106,58 @@ jobs:
...
+        run: |
+          echo "ARTIFACT_BUILD_NAME=opensearch-dashboards-${{ env.VERSION }}-${{ matrix.suffix }}.${{ matrix.ext }}" >> $GITHUB_ENV
+
+      - name: Run bootstrap
+        run: yarn osd bootstrap
+
+      - name: Build `${{ matrix.name }}`
+        run: yarn ${{ matrix.script }} --release
+
+      - uses: actions/upload-artifact@v3
```
Comment (ashwin-pc) [inline .github/workflows/build_and_test_workflow.yml:158]: Do we need do we need this step?

**Review by ashwin-pc — APPROVED** [review]
LGTM. just a few questions though.

**Review by Flyingliuhub — APPROVED** [review]

File: .github/workflows/build_and_test_workflow.yml (hunk 2 above)
Comment (kavilla) [reply .github/workflows/build_and_test_workflow.yml:158]: People can pull down their release distribution and start play with it. For future use, we can run some tests like BWC tests by pulling this artifact.

File: .github/workflows/build_and_test_workflow.yml (hunk 1 above)
Comment (kavilla) [reply .github/workflows/build_and_test_workflow.yml:142]: You are right we always take the latest to build. Perhaps we can have a follow-up to align the other jobs as well. But I can see a benefit to specify every time.

**Comment by kavilla** [comment]
Will merge in since main still on 2.0.

**Pull request merged**